src/lib/storage/constraints/table_key_constraint.hpp

```diff
@@ -51,6 +51,8 @@ class TableKeyConstraint final : public AbstractTableConstraint {
 
   /**
    * Returns whether or not the last validation and invalidation `CommitID`s tell us that this constraints is valid.
+   * Be aware that this does not mean that the constraint is valid in the current transaction, only that it was 
+   * validated last. To check if a constraint is guaranteed to be valid use `key_constraint_is_confidently_valid`.
```

Comment:
If I read that now, the name `is_valid()` pretty much should return true if sth is safely valid. The meaning here is that it was not invalidated after it was validated. Do you have any idea for a name describing that better?